test/integration/consul-container/test/multiport/explicit_destination_test.go

```diff
@@ -0,0 +1,273 @@
+			Group:        "catalog",
+			GroupVersion: "v2beta1",
+			Kind:         "Service",
+		},
+		Tenancy: &pbresource.Tenancy{
+			Partition: "default",
+			Namespace: "default",
+			PeerName:  "local",
+		},
+	}).WithData(t, &pbcatalog.Service{
+		Workloads: &pbcatalog.WorkloadSelector{Prefixes: []string{"static-server"}},
+		Ports: []*pbcatalog.ServicePort{
+			{TargetPort: "tcp", Protocol: pbcatalog.Protocol_PROTOCOL_TCP},
+			{TargetPort: "mesh", Protocol: pbcatalog.Protocol_PROTOCOL_MESH},
+		},
+	}).Write(t, resourceClient)
+
+	workloadPortMap := map[string]*pbcatalog.WorkloadPort{
+		"tcp": {
+			Port: 8080, Protocol: pbcatalog.Protocol_PROTOCOL_TCP,
+		},
+		"mesh": {
+			Port: 20000, Protocol: pbcatalog.Protocol_PROTOCOL_MESH,
+		},
+	}
+
+	rtest.ResourceID(&pbresource.ID{
+		Name: "static-server-workload",
+		Type: &pbresource.Type{
+			Group:        "catalog",
+			GroupVersion: "v2beta1",
+			Kind:         "Workload",
+		},
+		Tenancy: &pbresource.Tenancy{
+			Partition: "default",
+			Namespace: "default",
+			PeerName:  "local",
+		},
+	}).
+		WithData(t, &pbcatalog.Workload{
+			Addresses: []*pbcatalog.WorkloadAddress{
+				{Host: ipAddress},
+			},
+			Ports:    workloadPortMap,
+			Identity: "static-server-identity",
+		}).
+		Write(t, resourceClient)
+	return serverService
+}
+
+func createClientResources(t *testing.T, resourceClient *rtest.Client, staticServerRef *pbresource.Resource, ipAddress string) {
+	rtest.ResourceID(&pbresource.ID{
+		Name: "static-client-service",
+		Type: &pbresource.Type{
+			Group:        "catalog",
+			GroupVersion: "v2beta1",
+			Kind:         "Service",
+		},
+		Tenancy: &pbresource.Tenancy{
+			Partition: "default",
+			Namespace: "default",
+			PeerName:  "local",
+		},
+	}).WithData(t, &pbcatalog.Service{
+		Workloads: &pbcatalog.WorkloadSelector{Prefixes: []string{"static-client"}},
+		Ports: []*pbcatalog.ServicePort{
+			{TargetPort: "tcp", Protocol: pbcatalog.Protocol_PROTOCOL_TCP},
+			{TargetPort: "mesh", Protocol: pbcatalog.Protocol_PROTOCOL_MESH},
+		},
+	}).Write(t, resourceClient)
+
+	workloadPortMap := map[string]*pbcatalog.WorkloadPort{
+		"tcp": {
+			Port: 8080, Protocol: pbcatalog.Protocol_PROTOCOL_TCP,
+		},
+		"mesh": {
+			Port: 20000, Protocol: pbcatalog.Protocol_PROTOCOL_MESH,
+		},
+	}
+
+	rtest.ResourceID(&pbresource.ID{
+		Name: "static-client-workload",
+		Type: &pbresource.Type{
+			Group:        "catalog",
+			GroupVersion: "v2beta1",
+			Kind:         "Workload",
+		},
+		Tenancy: &pbresource.Tenancy{
+			Partition: "default",
+			Namespace: "default",
+			PeerName:  "local",
+		},
+	}).
+		WithData(t, &pbcatalog.Workload{
+			Addresses: []*pbcatalog.WorkloadAddress{
+				{Host: ipAddress},
+			},
+			Ports:    workloadPortMap,
+			Identity: "static-client-identity",
+		}).
+		Write(t, resourceClient)
+
+	destId := staticServerRef.GetId()
+	destRef := &pbresource.Reference{
+		Type:    destId.Type,
+		Tenancy: destId.Tenancy,
+		Name:    destId.Name,
+		Section: "",
+	}
```

Comment:
We can use `reference.Reference` to easily convert an ID into a reference.

Reply:
see below answer.

Reply:
yea, I think I am going to leave this as is.  I don't want to start changing import rules that someone else created to pull in source code to be used in the test.  I think pulling in other test code is okay in the case of a complex client that we are using from `resourcetest`.  I don't want to set the precedence here for something like this that is very simple.  thoughts?